src/lando/api/legacy/workers/landing_worker.py

```diff
@@ -143,6 +144,15 @@ def run_job(self, job: LandingJob) -> bool:
 
         job.transition_status(JobAction.LAND, commit_id=commit_id)
 
+        if job.is_pull_request_job:
+            # TODO: move this to different method, and retry if needed.
```

Comment:
This could be set via an async celery job.

Alternatively, could we use the `fixes:` keyword on the commit message? https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests, and let GitHub do the cleanup?